File: Python/SymmetricTreeTest.py
from unittest import TestCase

from SymmetricTree import SymmetricTree, TreeNode


class TestSymmetricTree(TestCase):
    def test_isSymmetric(self):
        st = SymmetricTree()

        self.assertTrue(st.isSymmetric(None))

        node0 = TreeNode(1)
        node0.left = TreeNode(2)
        node0.right = TreeNode(2)
        node0.left.left = TreeNode(3)
        node0.left.right = TreeNode(4)
        node0.right.left = TreeNode(4)
        node0.right.right = TreeNode(3)
        self.assertTrue(st.isSymmetric(node0))

        node1 = TreeNode(1)
        node1.left = TreeNode(2)
        node1.left.right = TreeNode(3)
        node1.right = TreeNode(2)
        node1.right.right = TreeNode(3)
        self.assertFalse(st.isSymmetric(node1))

File: Golang/SymmetricTree/SymmetricTree_test.go
package SymmetricTree

import "testing"

func TestIsSymmetric(t *testing.T) {

	if isSymmetric(nil) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isSymmetric(&TreeNode{
		Val: 1,
		Left: &TreeNode{
			Val: 2,
			Left: &TreeNode{
				Val: 3,
			},
			Right: &TreeNode{
				Val: 4,
			},
		},
		Right: &TreeNode{
			Val: 2,
			Left: &TreeNode{
				Val: 4,
			},
			Right: &TreeNode{
				Val: 3,
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isSymmetric(&TreeNode{
		Val: 1,
		Left: &TreeNode{
			Val: 2,
			Right: &TreeNode{
				Val: 3,
			},
		},
		Right: &TreeNode{
			Val: 2,
			Right: &TreeNode{
				Val: 3,
			},
		},
	}) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}
